src/main/java/org/folio/services/validator/engine/ValidationEngineServiceImpl.java

```diff
@@ -212,55 +208,54 @@ private void validatePasswordByRexExpRule(final String password,
     String remoteModuleUrl = okapiURL + rule.getImplementationReference();
 
     Promise<String> promise = Promise.promise();
-    HttpClientRequest passwordValidationRequest = httpClient.postAbs(remoteModuleUrl, validationResponse -> {
-      if (validationResponse.statusCode() == HttpStatus.SC_OK) {
-        validationResponse.bodyHandler(body -> {
-          String validationResult = body.toJsonObject().getString(ValidatorHelper.RESPONSE_VALIDATION_RESULT_KEY);
-          if (ValidatorHelper.VALIDATION_INVALID_RESULT.equals(validationResult)) {
-            errorMessages.add(rule.getErrMessageId());
-          }
-          promise.complete();
-        });
-      } else {
-        // TODO Inform administrator that remote module is down
-        logger.error("FOLIO module by the address " + remoteModuleUrl + " is not available.");
-        String errorMessage;
-        switch (rule.getValidationType()) {
-          case STRONG:
-            errorMessage = new StringBuilder()
-              .append("Programmatic rule ")
-              .append(rule.getName())
-              .append(" returns status code ")
-              .append(validationResponse.statusCode())
-              .toString();
-            logger.error(errorMessage);
-            promise.fail(errorMessage);
-            break;
-          case SOFT:
-            promise.complete();
-            break;
-          default:
-            errorMessage = "Please add an action for the new added " +
-              "rule type when internal FOLIO module is not available";
-            logger.error(errorMessage);
-            promise.fail(errorMessage);
-        }
-      }
-    });
+    HttpRequest<Buffer> passwordValidationRequest = webClient.postAbs(remoteModuleUrl);
     passwordValidationRequest
       .putHeader(OKAPI_HEADER_TOKEN, headers.get(OKAPI_HEADER_TOKEN))
       .putHeader(OKAPI_HEADER_TENANT, headers.get(OKAPI_HEADER_TENANT))
       .putHeader(HttpHeaders.CONTENT_TYPE.toString(), MediaType.APPLICATION_JSON)
       .putHeader(HttpHeaders.ACCEPT.toString(), MediaType.APPLICATION_JSON)
-      .end(buildResetPasswordAction(userId, password));
+      .sendJsonObject(buildResetPasswordAction(userId, password), ar -> {
+        if (ar.failed()) {
+          return;
```

Comment:
Should it fail the promise before returning?

Reply:
added, thank you.